### PR TITLE
Add options for square meter price

### PR DESCRIPTION
**Pull request by mictab**
I thought it might be interesting for a buyer to make sure that they don't pay too much for too little space, so I added options for specifying the maximum m2 price (and minimum) a buyer is willing to pay in rent. This is a completely optional filtering option which is only performed if specified by the user so it shouldn't interfere with the normal usage.

Also:

- Refactored the requirements to its own file and updated the README
- Rewrote the conditionals in the filtering for a more pythonic feel

Note: Right now it requires for a user to *both* specify a minimum and maximum m2 price. Would you like it some other way? 😄 (Granted that you'd like this PR, that is).

### CODE REVIEW

**Comment by Bridouille** [comment]
I didn't think about that filtering, I think it can be useful (for people who have a clear idea of what they looking for) 👍 !

Thank you for updating the README and the requirements.

I think we can add a default value if only one of them is provided ? So that : 

- we can omit the filter (not specifying any of them)
- having a max value (only specifying the max, setting the min to a very low default)
- having a min value (only specifying the min and setting the max to a very high value).

Basically the same thing I did with price and size.

**Comment by mictab** [comment]
Thanks for the feedback! Yes, if was considering that but I couldn't come up with any reasonable default values. Do you have any in mind?

**Comment by Bridouille** [comment]
I think a `DEFAULT_MIN_SQUARE_METER_PRICE` to 50 and `DEFAULT_MAX_SQUARE_METER_PRICE` to 1000 looks good.

Makes 1500 SEK / month for a 30m2 in the min case or 30 000 SEK / month for max case. Every apartment should be in between those haha

**Comment by mictab** [comment]
@Bridouille Should be fixed :)

**Comment by Bridouille** [comment]
Great work! Thanks for taking the time to add this, I appreciate it :)
